pipeline/load.py

```diff
@@ -0,0 +1,125 @@
+from os import environ
+import logging
+from dotenv import load_dotenv
+import pandas as pd
+from pymssql import connect, Connection, exceptions
+
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def get_connection() -> Connection:
+    """Connects to Microsoft SQL Server Database"""
+    logging.info("Attempting to connect to the database.")
+    try:
+        conn = connect(
+            server=environ["DB_HOST"],
+            port=environ["DB_PORT"],
+            user=environ["DB_USER"],
+            password=environ["DB_PASSWORD"],
+            database=environ["DB_NAME"],
+            as_dict=True
+        )
+        logging.info("Database connection successful.")
+        return conn
+    except KeyError as e:
+        logging.error("%s missing from environment variables.", e)
+        raise
+    except exceptions.OperationalError as e:
+        logging.error("Error connecting to database: %s", e)
+        raise
+    except Exception as e:
+        logging.error("Unexpected error while connecting to database: %s", e)
+        raise
+
+
+def get_botanists_id_mapping(connection: Connection, names: list) -> dict:
+    """Fetches botanist IDs for a list of names."""
+    logging.info("Fetching botanist IDs for names: %s", names)
+    query = f"""SELECT botanist_id, full_name FROM epsilon.botanist WHERE full_name IN ({
+        ', '.join(['%s'] * len(names))})"""
+    try:
+        with connection.cursor() as cur:
+            cur.execute(query, names)
+            result = cur.fetchall()
+            if not result:
+                logging.warning("No botanists found for the provided names.")
+            return {botanist['full_name']: botanist['botanist_id'] for botanist in result}
+    except exceptions.DatabaseError as e:
+        logging.error("Database error while fetching botanist IDs: %s", e)
+        raise
+    except Exception as e:
+        logging.error("Unexpected error while fetching botanist IDs: %s", e)
+        raise
+
+
+def get_plant_metric_data() -> pd.DataFrame:
+    """Reads the plant metric data from a CSV file."""
+    try:
+        df = pd.read_csv("./clean_plant_info.csv")
```

Comment:
Nice work, for a future pull request we can swap our reading a csv file to passing in the dictionary.